### ISSUE TITLE
Upgraded v 1 to 2.0 update_routines.php 

### ISSUE THREAD

**Issue by MikeyGMT** [e107inc/e107#1109]
/e107_admin/e107_update.php

System Debug
Update required in update_routines.php on line 1175 (Pages/Menus Table requires updating.)
Update required in update_routines.php on line 439 (Missing pref: admin_separate_plugins)

Update from 1.x to 2.0 > Update

Response > Success>Executed

Notice: Undefined variable: data, Line 190 of /home/xxx/public_html/e107_admin/update_routines.php

The code line is in function updateCore 

> >  $message = LAN_UPDATE_7." ".$data['title'];

But on line 180 reference to $data is commented out. 

> > //   foreach($this->core as $func => $data)

and also the closing } on line 210. 

Obviously I don't have enough knowledge to touch this part of the code.

Thanks

Mikey

**Comment by MikeyGMT** [comment]
Same problem on a second upgraded site. Noticed this in the Admin Error log

[03-Jul-2015 21:47:09 GMT] PHP Fatal error:  Allowed memory size of 33554432 bytes exhausted (tried to allocate 8195199 bytes) in /home/xxx/public_html/class2.php on line 2142

**Comment by MikeyGMT** [comment]
OK so I did this through phpMyAdmin
ALTER TABLE `e107_page` DROP `page_theme` ;

Then re-ran 
Update required in update_routines.php on line 1175 (Pages/Menus Table requires updating.)
Was no longer required.

Then I could update the core prefs


Then Forum Upgrade

Error
Creating table forum_new
Creating table forum_thread
Creating table forum_post
Creating table forum_track
Creation of table(s) failed. You can not continue until these are created successfully!

**Comment by MikeyGMT** [comment]
But I noticed ".MPREFIX." rather than #page

**Comment by MikeyGMT** [comment]
YayHay! Replaced X with ".MPREFIX." and it worked - tested on 2 upgraded sites. Committed that to my fork. 

	if($sql->field('page','page_theme') && $sql->gen("SELECT * FROM ".MPREFIX."page WHERE page_theme != '' AND menu_title = '' LIMIT 1"))
	{
		if ($just_check)
		{
			return update_needed("Pages/Menus Table requires updating.");	
		}
		
		if($sql->update('page',"menu_name = page_theme, menu_title = page_title, menu_text = page_text, menu_template='default', page_title = '', page_text = '' WHERE page_theme !='' AND menu_title = '' AND menu_text = '' "))
		{
			$sql->gen("ALTER TABLE `".MPREFIX."page` DROP `page_theme`");
			$mes = e107::getMessage();
			$log->addDebug("Successfully updated pages/menus table to new format. ");
		}
	
	}

**Comment by MikeyGMT** [comment]
Oh dear - I don't know then....

>> The use of MPREFIX is incorrect. There should be no field called e107_page_theme. There is obviously a problem, but this is not the correct fix for it. 

Here is my Table Structure
![image](https://cloud.githubusercontent.com/assets/2847651/8507821/ff38b842-224b-11e5-9efe-a8c565fecc1e.png)

It looks like it was used for Menu Name at some point. Example Menu.

![image](https://cloud.githubusercontent.com/assets/2847651/8507827/33bac9d4-224c-11e5-8917-df631c475149.png)

Menu Setup Page 
![image](https://cloud.githubusercontent.com/assets/2847651/8507835/d1e3f13a-224c-11e5-8ef6-c62227c1bda5.png)

This is in the DB 
![image](https://cloud.githubusercontent.com/assets/2847651/8507833/9187fb68-224c-11e5-89c7-63833b2410b7.png)

This is Site Version 

![image](https://cloud.githubusercontent.com/assets/2847651/8507839/251cfa86-224d-11e5-8e27-28d96780d9db.png)

**Comment by CaMer0n** [comment]
Thanks for this. I misread the code yesterday, but hopefully my commit fixes the issue with the page table.

**Comment by CaMer0n** [comment]
@MikeyGMT any news on this one?

**Comment by tgtje** [comment]
Just upgrading an existing v1 site (xampp running) for other purposes (translation related).
Using the latest master pack upgrade was difficult (forum related . issue> is installed but not in use <; step 3 can not complete unless the 2 existing forum tables in dbase are deleted)
But this post from@MikeyGMT is also actual.
soem debug info : 
Update required in update_routines.php on line 1193 (Pages/Menus Table requires updating.)
Update required in update_routines.php on line 444 (Missing pref: admin_separate_plugins)

The Menu table needs to have some paths corrected in its data.
LAN_UPDATE_31

This also points out (for me) AGAIN ( it looks) if there is ANY content in page_theme column, the 
step will not be completed.. ( i deleted content, and ran update > ok.

( the second line disappaers when closing dubber) all runs and is updated.

**Comment by MikeyGMT** [comment]
Hi Cameron
Sorry I missed the comment/fix, I will test soon. 
Mike

**Comment by tgtje** [comment]
@MikeyGMT ; i just the newer files already, it still happens...

**Comment by CaMer0n** [comment]
@tgtje Extensive work done on the Upgrade of core and forum. Please update files and let me know how it goes. Thanks

**Comment by tgtje** [comment]
@CaMer0n 

here you are (described as much as possible: just finished.

(using xampp; imported old dbase, using full v1 site files 3 languages installed)
Role play/Setting:
 Basic user which would update the site with the LATEST updatepack (github).

Nothing on site is changed, and ONLY the new v2 files are used to overwrite ALL.
Languages are NOT updated for the 2 foreign languages this case Dutch/German.
Site runs (as user normally would have i guess) Sitelanguage (this case Dutch -- v1 files).(so all is as is and untouched)

Files overwritten, no more no less
Running update  (LAN's are present instead of dutch txt, but that is normally against v2 use)
Following each step as displayed (delete files, delete plugins etc.. all done)
Running the update.

System is 'stuck' at LAN_UPDATE_4
using debugger (do not mind the language related lines > as above v1 vs v2)

NOTE : again the Page/Menu table is present.. (previous posts described)
again in this state looking at dbase, the page_theme column has content.
Sql deleted the content present.

Going back tp update
Click update and it start continuing.
From that point all goes on...
Update complete.
(of course when using languages one would need to update those too).
(movie is point in time at stuck point starting debugger)

https://youtu.be/W7TSViZqV3Q

**Comment by CaMer0n** [comment]
@tgtje are you able to send me an sql file by email? (You can remove all users except the first one) I need the v1 database to test the process.

**Comment by CaMer0n** [comment]
Please re-open if the issue continues.

**Comment by tgtje** [comment]
@CaMer0n hard to find a legal address, well, made a choice, it's on its way anyway (hope it arrives).

**Comment by tgtje** [comment]
@CaMer0n following exactly the same routine NOW with the newest ( fix ) file/s.
Success :+1:

**Comment by CaMer0n** [comment]
Thanks for all your help! :+1:

**Comment by upperking** [comment]
I cannot get past Creating table forum_new
Creating table forum_thread
Creating table forum_post
Creating table forum_track
Creation of table(s) failed. You can not continue until these are created successfully!

**Comment by tgtje** [comment]
Like in post https://github.com/e107inc/e107/issues/1012  more info please

**Comment by CaMer0n** [comment]
@upperking - please take a look at your table structure using phpmyadmin, it should provide clues.

**Comment by MikeyGMT** [comment]
If you have only one or two menus, a very quick workaround to this is to ...

1) Go to Menu's
2) Copy all Menu contents to a text file or to a page
3) Delete all Menus
4) Run the update and it will succeed.
5) Recreate your menus again

**Comment by MikeyGMT** [comment]
Downloaded the latest files today. Ran e107_update.php several times. I got a few LAN_UPDATE_31 errors but it seems to behave now. ( LAN_UPDATE_31 does not exist in lan_e107_update.php ).  
There are now no more pending updates.  Seems odd but I think it is solved.

**Comment by tgtje** [comment]
Have to note this one (was in older lan_e107_update php) 
define('LAN_UPDATE_31', 'Delete online_extended menu path');

Is not present anymore

**Comment by MikeyGMT** [comment]
OK, so it kept coming back. I did some debugging and found that online_extended_menu wasn't being deleted - it was set to not used (menu_location = 0) it looks like the just_check part was somehow skipping the removal of the row when we run the update.  

So if you keep getting the message:
> The Menu table needs to have some paths corrected in its data. 
Simply go to the menu table in php my admin and delete the row.

**Comment by CaMer0n** [comment]
..then it's a bug.

**Comment by MikeyGMT** [comment]
OK - so it kept re-appearing each time I went into Menu Manager. The latest git seems to delete the DB data successfully. I have now deleted the old core folder in the plugin directory. 

OK I should have done this clean up task some time back. However, there are 400+ old core files according to file inspector. It's going to be a time consuming task to delete all those old files, and on multiple installations. 

I will log a new enhancement for File Inspector, to add a function to select one or all old core files, zip them and then delete from the file system.
